src/main/scala/viper/silver/plugin/standard/inline/InlineRewrite.scala

```diff
@@ -0,0 +1,79 @@
+package viper.silver.plugin.standard.inline
+
+import viper.silver.ast.{Exp, Fold, LocalVar, Method, PredicateAccess, Program, Stmt, Unfold}
+
+trait InlineRewrite {
+
+  def inlinePredicates(method: Method, program: Program): Method = {
+    val expandedPres = method.pres.map { pre =>
+      expandPredicate(pre, program).fold(pre)(expandedPred => expandedPred)
+    }
+    val expandedPosts = method.posts.map { post =>
+      expandPredicate(post, program).fold(post)(expandedPred => expandedPred)
+    }
+    method.copy(name = method.name,
+      formalArgs = method.formalArgs,
+      formalReturns = method.formalReturns,
+      pres = expandedPres,
+      posts = expandedPosts,
+      body = method.body
+    )(pos = method.pos, info = method.info, errT = method.errT)
+  }
+
+  def removeUnfoldFold(method: Method, program: Program): Method = {
+    val expandablePredicateIds = getExpandablePredicateIds(method.pres, program)
+    val rewrittenBody = method.body.map { body =>
+      val bodyWithRemovedUnfolds = removeUnfolds(body.ss, expandablePredicateIds)
+      val bodyWithRemovedUnfoldFolds = removeFolds(bodyWithRemovedUnfolds, expandablePredicateIds)
+      body.copy(ss = bodyWithRemovedUnfoldFolds, scopedDecls = body.scopedDecls)(
+        pos = body.pos, info = body.info, errT = body.errT
+      )
+    }
```

Comment:
You also need `val expandablePostIds = getExpandablePredicateIds(method.posts, program)`, and then do `removeFolds(bodyWithRemovedUnfolds, expandablePostIds)` instead